### PR TITLE
CEPH-83575046 - Control functionality for RGW lifecycle

### PR DESCRIPTION
**Pull request by anrao19**
Signed-off-by: Anuchaithra Rao <anrao@redhat.com>

Set rgw_enable_lc_threads set to false and lc status in lc list for that bucket should change its state from unintial
but manual lc process for bucke should work

When invalid_date=false and rgw_enable_lc_threads =false: http://magna002.ceph.redhat.com/ceph-qe-logs/Anuchaithra/CEPH-83575046.log
When invalid_date=false and rgw_enable_lc_threads =true: http://magna002.ceph.redhat.com/ceph-qe-logs/Anuchaithra/CEPH-83575046_1.log
When invalid_date=true http://magna002.ceph.redhat.com/ceph-qe-logs/Anuchaithra/CEPH-83575046_2.log

### CODE REVIEW

**Review by ckulal — APPROVED** [review]
lgtm

File: rgw/v2/tests/s3_swift/test_bucket_lifecycle_object_expiration.py [hunk 1]
```diff
@@ -181,11 +190,57 @@ def test_exec(config):
...
+                        )
+                        list_lc_after = json.loads(
+                            utils.exec_shell_cmd("radosgw-admin lc list")
+                        )
+                        log.info(f"lc lists is {list_lc_after}")
+                        for data in list_lc_after:
+                            if data["bucket"] == bucket.name:
+                                if data["status"] == "UNINITIAL":
+                                    raise TestExecError(
+                                        f"Even if rgw_enable_lc_threads set to false manual lc process for bucket {bucket.name} should worker"
```
Comment (udaysk23) [inline rgw/v2/tests/s3_swift/test_bucket_lifecycle_object_expiration.py:241]: Please fix the typo

Comment (anrao19) [reply rgw/v2/tests/s3_swift/test_bucket_lifecycle_object_expiration.py]: Made changes

**Review by viduship — APPROVED** [review]
looks good

**Review by mkasturi18 — APPROVED** [review]

**Pull request merged**